src/lua/protobuf_wireformat.lua

```diff
@@ -97,6 +141,29 @@ local function encode_varint(value)
     return ffi.string(buf, size)
 end
 
+-- Decode input value as VARINT without a tag.
+--
+-- Input value type: string
+--
+-- This is a helper function to decode tag and data values.
+local function decode_varint(ctx)
+    local msb
+    local res = 0
+    local len = 0
+    repeat
+        local payload = ffi.cast('uint64_t', string.byte(ctx.val, ctx.pos))
+        if len >= 10 or payload == nil then
+            error('Incorrect msb for varint to decode')
```

Comment:
Imagine that you have a 100 KiB data and got this error on decoding. Your actions?

Sometimes things are even more complicated: you have no the message, just the error that the decoder reported.

The message now says not more than 'something is bad', at least for me. Can we at least say 'a varint exceeds 10 bytes is size' -- this is alredy something.

Next, we have the decoding context and we definitely can report the position, where the decoding fails. I suggest to consider introducing a function that raises decoding error with extending the given error message with extra infornation from the decoding context.

I guess we can even save a path to the current field (like `foo.bar.baz`) to the decoding context and report it in the error.

Reply:
I want to work on specific function `wrap_error` which will proceed generation of error message based on the context from which it was called. But it's implementation isn't ready yet.